api/events/types.go

```diff
@@ -129,14 +129,22 @@ const (
 
 type Range struct {
 	Unit RangeType
-	From uint64
-	To   uint64
+	From *uint64 `json:"from,omitempty"`
+	To   *uint64 `json:"to,omitempty"`
 }
 
 func ConvertRange(chain *chain.Chain, r *Range) (*logdb.Range, error) {
 	if r == nil {
 		return nil, nil
 	}
+	if r.From == nil {
+		from := uint64(0)
+		r.From = &from
+	}
+	if r.To == nil {
+		to := uint64(math.MaxUint64)
+		r.To = &to
```

Comment:
Is it correct to say that previous if the to was nil it would be set to 0 and now it will be set to MaxUint64 ? 
If so, could that create some issue ?

Reply:
It depends.

If the `range` object was completely omitted, the entire chain would be queried.

If the `range` was set, but not the `to`, it would default to 0, which is different behaviour to the above. So this sets the `to` to the max possible. But it gets set to the max of that and the chain head below

Reply:
Yeah, that's a small wrinkle, but I think that's fine. No massive change in expected behaviour.